src/probnum/filtsmooth/statespace/discrete_transition.py

```diff
@@ -35,7 +35,7 @@ def __init__(self, dynamicsfun, diffmatfun, jacobfun=None):
         self._diffmatfun = diffmatfun
         self._jacobfun = jacobfun
 
-    def transition_realization(self, real, start, stop=None):
+    def transition_realization(self, real, start, stop=None, **kwargs):
```

Comment:
Why did you add a `**kwargs` here, was there some explicit reason? If yes we should keep track of that to understand if we want to have them around and when. 

And please don't get me wrong, I hope I'm not too annoying with those comments^^ If you think that it's best right now to have them around that's fine for me! I'm just a bit confused since I don't have the impression that the majority of our code even has them in function signatures, so I would personally refrain from adding new instances.

Reply:
I added them here, because they capture some of those extra arguments (e.g., `step`). Without, it is not possible to combine the two `generate` functions into one. 

No worries :) I think it is good to question those kwargs!

Reply:
Cool, then of course it's totally appropriate to use it :)